app/liquid_tags/asciinema_tag.rb

```diff
@@ -18,14 +19,15 @@ def render(_context)
   private
 
   def parse_id(input)
-    input_no_space = input.delete(" ")
-    raise StandardError, "Invalid Asciinema ID: {% asciinema #{input_no_space} %}" unless valid_id?(input_no_space)
-
-    input_no_space
+    sanitized_input = input.strip
+    match_data = sanitized_input.match(ASCIINEMA_URL_REGEX)
+    match_data ? match_data["id"] : validate(sanitized_input)
   end
 
-  def valid_id?(id)
-    id.match?(/\A\d+\Z/)
+  def validate(id)
+    raise "Invalid Asciinema ID: #{id}" unless id.match?(/\A\d+\z/)
```

Comment:
Changed this from `\Z` to `\z`, since the former would accept a trailing newline, which we don't want.